Automattic/jetpack
# Add Jetpack Modules redux store to shared utils

Fixes https://github.com/Automattic/vulcan/issues/12

## Proposed changes:

In order to have a consistent state of Modules statuses across Monorepo I think it would be beneficial to have a Modules Redux store. This eventually could replace (or enhance) hook `useModuleStatus` and give more information. 

- Creating Redux store for general information on Jetpack modules inside `shared-extension-utils`
- Use selector and `dispatch` action to control the module state within Form package (and block)
- Update `withAiAssistantComponents` HOC to use `useIsPossibleToExtendJetpackFormBlock` hook to have reactive data

### Other information:


## Jetpack product discussion

* Epic: #32401
## Does this pull request change what data or activity we track or use?



## Testing instructions:






- Clone this branch locally. You can run it with your Jurassic Tube setup.
- In `/wp-admin/admin.php?page=jetpack_modules` verify that Contact Form module is disabled
- Create new post `/wp-admin/post-new.php` and add new block

<img width="517" alt="Screenshot at Oct 11 11-48-04" src="https://github.com/Automattic/jetpack/assets/60262784/dbed5b2e-2063-43c1-b28b-4256947e062c">

- Verify that no AI assistant present, when block is disabled
- Activate module. Verify that straight away the AI assistant is available

<img width="403" alt="Screenshot at Oct 11 14-33-46" src="https://github.com/Automattic/jetpack/assets/60262784/2f59dadd-99d6-490d-93ea-c51cf400927d">

- Check any additional implications you could think of with the module or AI assistatnt
